restlicsharpclient/restliclient/transport/RestliCallbackAdapter.cs

```diff
@@ -39,8 +39,17 @@ public RestliCallbackAdapter(RestResponseDecoder<TResponse> responseDecoder, Res
         public void OnSuccess(HttpResponse httpResponse)
         {
             TransportResponse transportResponse = new TransportResponse(httpResponse);
-            TResponse response = responseDecoder.DecodeResponse(transportResponse);
-            callback.OnSuccess(response);
+
+            if (transportResponse.hasError())
+            {
+                ClientErrorResponse clientErrorResponse = transportResponse.getError();
+                callback.OnError(clientErrorResponse);
+            }
+            else
+            {
+                TResponse response = responseDecoder.DecodeResponse(transportResponse);
+                callback.OnSuccess(response);
+            }
         }
 
         public void OnError(HttpResponse httpResponse)
```

Comment:
This seems not used at all. I am suggesting that we can change TransportCallback interface only have one method: onResponse(HttpResponse).